### ISSUE TITLE
🐛Bugs in run_tf_ner.py

### ISSUE THREAD

**Issue by jia-zhuang** [huggingface/transformers#3389]
Found a bug in [run_tf_ner.py](https://github.com/huggingface/transformers/blob/master/examples/ner/run_tf_ner.py) at line 170 and 325:

```python
loss_fct = tf.keras.losses.SparseCategoricalCrossentropy(reduction=tf.keras.losses.Reduction.NONE)
```
refer to [tf.keras.losses.SparseCategoricalCrossentropy](https://www.tensorflow.org/api_docs/python/tf/keras/losses/SparseCategoricalCrossentropy)

```python
tf.keras.losses.SparseCategoricalCrossentropy(
    from_logits=False, reduction=losses_utils.ReductionV2.AUTO,
    name='sparse_categorical_crossentropy'
)
```

> `from_logits`: Whether y_pred is expected to be a logits tensor. By default, we assume that y_pred encodes a probability distribution. Note: Using from_logits=True may be more numerically stable.

So I think `loss_fct` should be init with `from_logits=True` if the `TFBertForTokenClassification` just return pure logits rather than softmax output.

**Comment by jplu** [comment]
Hey @jia-zhuang,

As you can see [here](https://github.com/huggingface/transformers/blob/master/examples/ner/run_tf_ner.py#L525) I update the classification layer to add a softmax as activation, then the `from_logits=True` is not necessary.

**Comment by jia-zhuang** [comment]
@jplu Thanks for your reply! I learn a lot from your code.
